### PR TITLE
Added BookController, BookDto and CreateBookRequestDto classes. Also …

### PR DESCRIPTION
**Pull request by sovisou**
…implemented mappers and Mapstruct. Made some changes in service and repository.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,43 +28,78 @@
...
-			<version>8.0.33</version>
+			<groupId>org.apache.commons</groupId>
+			<artifactId>commons-dbcp2</artifactId>
+			<version>2.12.0</version>
 		</dependency>
 		<dependency>
-			<groupId>org.projectlombok</groupId>
-			<artifactId>lombok</artifactId>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-orm</artifactId>
```
Comment (Elena-Bruyako) [inline pom.xml]: why do you need this dependency?

File: pom.xml [hunk 2]
```diff
@@ -28,43 +28,78 @@
...
-			<artifactId>spring-boot-starter-data-jpa</artifactId>
+			<artifactId>spring-boot-starter-test</artifactId>
+			<scope>test</scope>
 		</dependency>
 		<dependency>
-			<groupId>mysql</groupId>
-			<artifactId>mysql-connector-java</artifactId>
-			<version>8.0.33</version>
+			<groupId>org.apache.commons</groupId>
+			<artifactId>commons-dbcp2</artifactId>
```
Comment (Elena-Bruyako) [inline pom.xml]: why do you need this dependency?

File: pom.xml [hunk 3]
```diff
@@ -28,43 +28,78 @@
...
+		<dependency>
+			<groupId>org.mapstruct</groupId>
+			<artifactId>mapstruct-processor</artifactId>
+			<version>${mapstruct.version}</version>
+			<scope>provided</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok-mapstruct-binding</artifactId>
+			<version>0.2.0</version>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
			<version>${lombok.mapstruct.binding.version}</version>
```
add version to properties

File: src/main/java/com/onlinebookstore/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

File: src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -46,4 +47,12 @@ public List<Book> findAll() {
             throw new DataProcessingException("Can`t find books in the DB", e);
         }
     }
+
+    @Override
+    public Optional<Book> findBookById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Book book = session.find(Book.class, id);
+            return Optional.ofNullable(book);
+        }
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java:58]: add catch block

File: src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -46,4 +47,12 @@ public List<Book> findAll() {
             throw new DataProcessingException("Can`t find books in the DB", e);
         }
     }
+
+    @Override
+    public Optional<Book> findBookById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Book book = session.find(Book.class, id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/repository/impl/BookRepositoryImpl.java]: inline with return

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by ahoienko — APPROVED** [review]
Good job!

**Pull request merged**